app/controllers/users_controller.rb

```diff
@@ -17,9 +19,41 @@ def create
       render :new
     end
   end
+  def edit; end
+  def update
+    if @user.update(user_params)
+      flash[:success] = t("profile_update")
+      redirect_to @user
+    else
+      render "edit"
+    end
+  end
+  def destroy
+    if @user&.destroy
```

Comment:
chức năng xóa ko cần check quyền admin ah? 

Ngoài ra, cần `&.` ko em

Reply:
Ở view/users/_user.html.erb
<% if current_user.admin? && !current_user?(user) %>
   <%= link_to t("delete"), user, method: :delete, data: {confirm: t("you_sure")} %>
<% end %>
Nếu user hiện tại là admin thì mới hiện nút delete sử dụng phương thức delete. Làm như vậy có được ko chị?

Reply:
nhưng nếu ng dùng giả lập và pass view thì sao

phía client thì lúc nào cũng có thể giả lập request đồ dc. nên đảm bảo an toàn, luôn bắt ở server nhé 

ở client ý nghĩa là thân thiện ng dùng thôi, ng dùng có thể cảm nhận mình có thẻ tương tác trực tiếp đến sever 
:v: